### PR TITLE
Check unique values of site_pop per site

### PR DESCRIPTION
**Pull request by gvegayon**
This pull request includes a change enhancing data validation: checks for unique site populations. 

### Enhancements to data validation:

* Added a check to ensure that `site_pop` is unique per site in the `validate_ww_conc_data` function (`R/validate.R`).
* Introduced a new test to verify that the population size per site remains constant (`tests/testthat/test_preprocess_ww_data.R`).

### Documentation updates:

* Updated `NEWS.md` to include a note about the new check for unique `site_pop` values.

### Misc

* Adds the `dplyr::` prefix in multiple files (by @kaitejohnson):
  * `flag_ww_outliers` function in `R/preprocessing.R`
  * `test_preprocess_ww_data.R` tests

### CODE REVIEW

File: R/validate.R [hunk 1]
```diff
@@ -80,6 +80,28 @@ validate_ww_conc_data <- function(ww_data,
...
+      " Got site(s) with more than one population: ",
+      paste0(
+        records_per_site_per_pop$site[records_per_site_per_pop$n > 1],
+        " (",
+        records_per_site_per_pop$n[records_per_site_per_pop$n > 1],
+        " records)",
+        collapse = ", "
+      )
+    )
+  }
```
Comment (gvegayon) [inline R/validate.R:102]: This is the check I'm implementing, @kaitejohnson. LMK if this makes sense to you. The CI is not passing.

Comment (kaitejohnson) [reply R/validate.R:102]: Cool looks like the test of preprocess ww data is failing -- I will investigate!

Comment (kaitejohnson) [reply R/validate.R:102]: Yep - its failing bc I actually didn;t change the order of the site pops when I changed the order of the sites to be scrambled (basically it failed for the right reason, and hadn't been caught before)

Comment (kaitejohnson) [reply R/validate.R:102]: Just adjusted (hope its ok that I pushed to your branch directly...)

**Comment by kaitejohnson** [comment]
Pending CI passes, this LGTM, TY @gvegayon !

**Review by kaitejohnson — APPROVED** [review]

**Pull request merged**